### PR TITLE
DISTANCE_SENSOR adopting

### PR DESCRIPTION
**Pull request by TSC21**
[work in progress] This PR pretends to add `DISTANCE_SENSOR` msg as the main message for pub range sensor data, being altimeters or side sensors for obsavoid.

### CODE REVIEW

File: src/modules/uORB/topics/distance_sensor.h [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+ */
+
+/**
+ * Distance sensor data, with range in centimeters (not in SI).
+ *
+ * @see http://en.wikipedia.org/wiki/International_System_of_Units
+ */
+struct distance_sensor_s {
+
+	uint32_t time_boot_ms; /**< Time since system boot */
```
Comment (LorenzMeier) [inline src/modules/uORB/topics/distance_sensor.h]: Please put this into the msg folder and let it auto-generate.

**Comment by TSC21** [comment]
@LorenzMeier can you please do a mavlink release so I can add `msg.type = MAV_DISTANCE_SENSOR_INFRARED` to `mavlink/mavlink_messages.cpp`? Thanks

**Comment by mhkabir** [comment]
@TSC21 Can you please remove the auto-generated uorb topic file. :)

**Comment by TSC21** [comment]
Remove? Lorenz asked me in a comment to add it :D

**Comment by vooon** [comment]
@TSC21 @mhkabir mean remove .h-file, now it generates from .msg-file.

**Comment by TSC21** [comment]
OK I understood now. I was looking the `.h` in the repo vs the ones generated and checked that the autogenerated ones are not in the repo. Thanks for warning me ;)

**Comment by LorenzMeier** [comment]
I updated the mavlink lib on master.

**Comment by TSC21** [comment]
@LorenzMeier thanks!

File: msg/distance_sensor.msg [hunk 2]
```diff
@@ -0,0 +1,17 @@
+# DISTANCE_SENSOR message data
+
+uint32 time_boot_ms
+
+uint16 min_distance
```
Comment (jgoppert) [inline msg/distance_sensor.msg]: units?

**Comment by vooon** [comment]
Also there question about id:
As i know now firmware uses several topics for imu, may i be useful to split data by id?
E.g. distance_sensor0, distance_sensor1 ... distance_sensorN. N = id.

File: msg/distance_sensor.msg (hunk 2 above)
Comment (TSC21) [inline msg/distance_sensor.msg]: centimeters

**Comment by TSC21** [comment]
> Also there question about id:
> As i know now firmware uses several topics for imu, may i be useful to split data by id?
> E.g. distance_sensor0, distance_sensor1 ... distance_sensorN. N = id.

@vooon I second that idea.

**Comment by TSC21** [comment]
@LorenzMeier I'm getting this on the CI verification:
```
@ clang++ -MD -c -fno-exceptions -fno-rtti -std=c++0x -fno-threadsafe-statics -D__CUSTOM_FILE_IO__ -g -Wall -Wextra -Werror -Wfloat-equal -Wpointer-arith -Wmissing-declarations -Wpacked -Wno-unused-parameter -Wno-packed -Werror=format-security -Werror=array-bounds -Wfatal-errors -Werror=unused-variable -Werror=reorder -Werror=uninitialized -Werror=init-self -Wno-gnu-array-member-paren-init -Wno-missing-field-initializers -Os -g3 -fno-strict-aliasing -fomit-frame-pointer -funsafe-math-optimizations -fno-builtin-printf -ffunction-sections -fdata-sections -DCONFIG_ARCH_BOARD_POSIXTEST -D__PX4_LINUX -D__PX4_POSIX -Dnoreturn_function= -I/home/travis/build/PX4/Firmware//src/modules/systemlib -I/home/travis/build/PX4/Firmware//src/lib/eigen -I/home/travis/build/PX4/Firmware//src/platforms/posix/include -I/home/travis/build/PX4/Firmware//mavlink/include/mavlink -Wno-error=shadow -DCONFIG_WCHAR_BUILTIN -Wno-effc++ -I/home/travis/build/PX4/Firmware/src/ -I/home/travis/build/PX4/Firmware/src//modules/ -I/home/travis/build/PX4/Firmware/src/include/ -I/home/travis/build/PX4/Firmware/src/lib/ -I/home/travis/build/PX4/Firmware/src/platforms/ -I/home/travis/build/PX4/Firmware/src/drivers/boards/posix -Wno-deprecated-register -Wno-tautological-constant-out-of-range-compare -Wno-unused-private-field -Wno-unused-const-variable -fvisibility=hidden -include /home/travis/build/PX4/Firmware/src/include/visibility.h /home/travis/build/PX4/Firmware/src/modules/systemlib/circuit_breaker.cpp -o circuit_breaker.cpp.o
CC:      /home/travis/build/PX4/Firmware/src/modules/systemlib/circuit_breaker_params.c
make[3]: *** No rule to make target `/home/travis/build/PX4/Firmware/Build/git_version.c', needed by `/home/travis/build/PX4/Firmware/Build/git_version.c.o'.  Stop.
make[3]: Leaving directory `/home/travis/build/PX4/Firmware/Build/posix_default.build/home/travis/build/PX4/Firmware/src/modules/systemlib'
make[2]: *** [/home/travis/build/PX4/Firmware/Build/posix_default.build//home/travis/build/PX4/Firmware/src/modules/systemlib/module.pre.o] Error 2
make[2]: Leaving directory `/home/travis/build/PX4/Firmware/Build/posix_default.build'
make[1]: *** [/home/travis/build/PX4/Firmware/Build/posix_default.build/firmware.a] Error 2
make[1]: Leaving directory `/home/travis/build/PX4/Firmware'
make: *** [posix] Error 2

The command "make posix -j4" exited with 2.
```
Which is not directly related with the changes made in this commit. Is there any problem you can point out on `master` repo?

**Comment by TSC21** [comment]
I'm closing this PR, as https://github.com/PX4/Firmware/pull/2224 is the current one!

**Comment by bansiesta** [comment]
@Zefz and @TSC21: thanks for the work here. I made a bunch of changes, mostly to the LidarLitePWM, and finally got the PWM driver running in #2224 which is now merged.

**Pull request closed without merging**